packages/next/src/server/app-render/required-scripts.tsx

```diff
@@ -5,24 +5,35 @@ import ReactDOM from 'react-dom'
 export function getRequiredScripts(
   buildManifest: BuildManifest,
   assetPrefix: string,
+  crossOrigin: string | undefined,
   SRIManifest: undefined | Record<string, string>,
   qs: string,
   nonce: string | undefined
-): [() => void, string | { src: string; integrity: string }] {
+): [
+  () => void,
+  { src: string; integrity?: string; crossOrigin?: string | undefined }
+] {
   let preinitScripts: () => void
   let preinitScriptCommands: string[] = []
-  let bootstrapScript: string | { src: string; integrity: string } = ''
+  const bootstrapScript: {
+    src: string
+    integrity?: string
+    crossOrigin?: string | undefined
+  } = {
+    src: '',
+    crossOrigin,
+  }
```

Comment:
Although the `bootstrapScripts` option of `ReactDOMServer.renderToReadableStream` supports both string and object type, here I enforce the use of object type so we won't forget other attributes in the future.